src/dashboard/src/middleware/common.py

```diff
@@ -65,3 +66,28 @@ def process_exception(self, request, exception):
             return
         if isinstance(exception, self.EXCEPTIONS):
             return render(request, 'elasticsearch_error.html', {'exception_type': str(type(exception))})
+
+
+SHIBBOLETH_REMOTE_USER_HEADER = getattr(
+    settings, 'SHIBBOLETH_REMOTE_USER_HEADER', 'REMOTE_USER'
+)
+
+
+class CustomShibbolethRemoteUserMiddleware(ShibbolethRemoteUserMiddleware):
+    """
+    Custom version of Shibboleth remote user middleware
+
+    THe aim of this is to provide a custom header name that is expected
+    to identify the remote Shibboleth user
+    """
+    header = SHIBBOLETH_REMOTE_USER_HEADER
+
+    def make_profile(self, user, shib_meta):
+        """
+        Customize the user based on shib_meta mappings (anything that's not
+        already covered by the attribute map)
+        """
+        # Make the user an administrator if they are in the designated admin group
+        entitlements = shib_meta['entitlement'].split(';')
```

Comment:
`make_profile` runs every time a user logs into the app, and keeps them updated with details from Shibboleth. (so names will automatically change). Here, we keep the permissions up to date too. This allows for the possibility that a user might lose or gain admin entitlement.